Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -2305,6 +2321,26 @@ bool RenderLayerBacking::updateMaskingLayer(bool hasMask, bool hasClipPath)
     return layerChanged;
 }
 
+void RenderLayerBacking::updateChildClippingStrategy(bool needsDescendantsClippingLayer)
+{
+    auto needsClipMaskLayer = [&] {
+        return needsDescendantsClippingLayer && is<RenderBox>(renderer()) && (renderer().style().hasBorderRadius() || renderer().style().clipPath());
```

Comment:
This code has a problem.  This will create an unnecessary child clipping mask layer.
The original code created the child clipping mask layer only if GraphicsLayer::supportsRoundedClip() returned false.
https://github.com/WebKit/WebKit/blob/6d4c5f059b8767e4b3a7696a94a90660bb411048/Source/WebCore/rendering/RenderLayerBacking.cpp#L2327
And, GraphicsLayer::supportsRoundedClip()  never returned false for all port.

Reply:
Hello! 

Thanks for the review. This is exactly why I didn't move towards a direct revert, and opted to have it run through EWS completely first. Also, it wasn't a clean and easy revert. It had some conflicts and I tried to deal with them as best as I could.